### PR TITLE
filetype extractor

### PR DESCRIPTION
**Pull request by Ahmed-Badr01**
#272 

PR checklist:
- [ ] Tested by creator locally
- [ ] Tested by creator on remote App
- [ ] Tested by reviewer locally
- [ ] Tested by reviewer on remote App
- [ ] Conforms with the agreed upon code pattern

### CODE REVIEW

**Review by LeonardPuettmannKern — CHANGES_REQUESTED** [review]
Looks great, thanks for contributing! I've added a suggestion for the code_snippet.md. Apart from that it looks good. 👍

File: extractors/python_functions/filetype_extraction/code_snippet.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        for pattern, filetype in zip(patterns, file_types_json):
+            for match in pattern.finditer(text):
+                try:
+                    start, end = match.span()
+                    span = record[YOUR_ATTRIBUTE].char_span(start, end, alignment_mode="expand")
+                except AttributeError:
+                    break
+                filetypes[f"{filetype[1:]}"].append([span.start, span.end])
+
+        return filetypes
```
[the remark this answers is not in the document]
Comment (Ahmed-Badr01) [reply extractors/python_functions/filetype_extraction/code_snippet.md:28]: Actually, I do. I'm not sure if you are on Twitter or maybe even Skype, but I'm thinking about making another extractor, and I need to ask some questions about your comment and changes before I do that.

Let me know if either platform is an option. I currently have no presence on Discord or LinkedIn. Thank you in advance.

[the remark this answers is not in the document]
Comment (LeonardPuettmannKern) [reply extractors/python_functions/filetype_extraction/code_snippet.md:28]: Hi @Ahmed-Badr01 , I sadly don't have Twitter or Skype, but maybe we can just have a call over Google Meets? We've recently released a new version of bricks which changes some of the structure. If you like, I can explain everything to you then. 👍

[the remark this answers is not in the document]
Comment (Ahmed-Badr01) [reply extractors/python_functions/filetype_extraction/code_snippet.md:28]: Hello, Leonard.

Google Meets sounds fine, let's have a call sometime.

**Pull request closed without merging**